oanda_v20_platform/oanda/oanda.py

```diff
@@ -0,0 +1,399 @@
+
+import os
+import requests
+import json
+import time
+from notifier.sms import TwilioSMS
+from decouple import config
+import logging
+
+
+class Base(object):
+    """Base class to access account, and facilitate inheritance of the subclasses
+    """
+    # TODO test if this base class is still required?
+    def __init__(self, **kwargs): pass
+
+class Oanda(Base):
+    """Sets up access to an Oanda account and the market data stream for the chosen asset 
+    e.g. currency pair or commodity. 
+
+        Args:
+            token (str, required): 
+                The Oanda API token. Defaults to env variables.
+            account (str, required): 
+                The Oanda account number. Defaults to env variables.
+            practice (bool, required): 
+                Use the Oanda practice stream or go live with real money. Defaults to True.
+            pair (str, required): 
+                Which asset to trade. Defaults to 'EUR_USD'.
+            text_notifications (bool, optional): 
+                If sms notifications are set up do you want to use them?. Defaults to False.
+        """
+    
+    def __init__(self, token=config('PRACTICE_TOKEN'), account=config('PRACTICE_ACCOUNT'), practice=True, pair='EUR_USD', 
+                text_notifications=False, **kwargs):
+        
+
+        super().__init__(**kwargs)
+        self.logger = logging.getLogger(__name__)
+        self.token = token
+        self.account = account
+        self.practice = practice
+        self.pair = pair
+        self.text_notifications = text_notifications
+        self.base_url = self.set_base_url()
+        self.headers = self.set_headers()
+
+
+    def set_base_url(self):
+        if self.practice:
+            return 'https://api-fxpractice.oanda.com'
+        else:    
+            return 'https://stream-fxtrade.oanda.com'
+
+    
+    def set_headers(self):
+        return {'Authorization': 'Bearer ' + self.token}
+
+    def print_details(self):
+        print(f'Token: {self.token}')
+        print(f'Account Number: {self.account}')
+        print(f'Base Url: {self.base_url}')
+        print(f'Headers: {self.headers}')
+
+
+class Account(Oanda):
+    """Gets the current account status and attributes.
+    Subclass of Oanda.
+    """
+    def __init__(self, **kwargs):
+
+        super().__init__(**kwargs)
+        self.get_account()
+        self.get_account_balance()
+        
```

Comment:
The **kwargs combined with the named keywords in the class that is using them is the trick. The subclasses can access the attributes and methods of the parent classes easily, super handles passing what is needed. see the 
screenshot of an instance of DataFeed() you can see that it could use anything from the classes it inherits from.
![Screenshot (29)](https://user-images.githubusercontent.com/12174019/118727513-c5b45880-b82a-11eb-874d-dc10b03d8fcb.png)

Whereas an instance of Account can only get whats it inherits from:
![Screenshot (30)](https://user-images.githubusercontent.com/12174019/118728511-3314b900-b82c-11eb-85e9-dc70bdba8794.png)

From memory, I think you were trying to pass stuff the other way from DataFeed up to the parents

Reply:
I was trying to inherit from the parent to the child, but I probably had it backwards. hahaha thanks for the explanation. I wasn't using kwargs either as part of the inheritance.  I'll build a small dummy project to practice this more.  Frankly, class inheritance is my weakest skill in python cuz I come from a node background and classes are rarely used. I'm still learning that and need more practice.